main.go

```diff
@@ -66,53 +72,117 @@ const (
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cipherNames, minTypedTLSVersion := ocpconfigv1.TLSProfiles[tlsProfile.Type].Ciphers, ocpconfigv1.TLSProfiles[tlsProfile.Type].MinTLSVersion
+	cfg.CipherSuites = common.CipherIDs(cipherNames, &setupLog)
+	cfg.MinVersion, err = crypto.TLSVersion(string(minTypedTLSVersion))
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+type prometheusServer struct {
+	server   *http.Server
+	client   client.Client
+	certPath string
+	keyPath  string
+}
+
+// InjectClient implements the interface that is used to inject
+// the Client dependency initialized by the ControllerManager.
+func (s *prometheusServer) InjectClient(client client.Client) error {
+	s.client = client
+	return nil
+}
+
+// NeedLeaderElection implements the LeaderElectionRunnable interface, which indicates
+// the prometheus server doesn't need leader election.
+func (s *prometheusServer) NeedLeaderElection() bool {
+	return false
+}
+
+func (s *prometheusServer) Start(ctx context.Context) error {
 	go func() {
-		err := server.ListenAndServeTLS(path.Join(sdkTLSDir, sdkTLSCrt), path.Join(sdkTLSDir, sdkTLSKey))
+		err := s.server.ListenAndServeTLS(s.certPath, s.keyPath)
 		if err != nil {
 			setupLog.Error(err, "Failed to start Prometheus metrics endpoint server")
 		}
 	}()
 	return nil
 }
 
-func getWebhookServer(sspTLSOptions common.SSPTLSOptions) *webhook.Server {
-	// If TLSSecurityProfile is empty, we want to return nil so that the default
-	// webhook server configuration is used.
-	if sspTLSOptions.IsEmpty() {
-		return nil
+func (s *prometheusServer) getPrometheusTLSConfig(ctx context.Context, certWatcher *certwatcher.CertWatcher) *tls.Config {
+	return &tls.Config{
+		GetConfigForClient: func(_ *tls.ClientHelloInfo) (*tls.Config, error) {
+			cfg := &tls.Config{}
+			cfg.GetCertificate = certWatcher.GetCertificate
+			return getConfigForClientCallback(cfg, s.client, ctx)
+		},
+	}
+}
+
+func getPrometheusServer(metricsAddr string, ctx context.Context) (*prometheusServer, error) {
+	setupLog.Info("Starting Prometheus metrics endpoint server with TLS")
+	metrics.Registry.MustRegister(common_templates.CommonTemplatesRestored)
+	metrics.Registry.MustRegister(common.SSPOperatorReconcileSucceeded)
+	handler := promhttp.HandlerFor(metrics.Registry, promhttp.HandlerOpts{})
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", handler)
+
+	metricsServer := &prometheusServer{}
+
+	metricsServer.certPath = path.Join(sdkTLSDir, sdkTLSCrt)
+	metricsServer.keyPath = path.Join(sdkTLSDir, sdkTLSKey)
+
+	certWatcher, err := certwatcher.New(metricsServer.certPath, metricsServer.keyPath)
+	if err != nil {
+		return nil, err
 	}
 
-	tlsCfgFunc := func(cfg *tls.Config) {
-		cfg.CipherSuites = sspTLSOptions.CipherIDs(&setupLog)
-		setupLog.Info("Configured ciphers", "ciphers", cfg.CipherSuites)
+	go func() {
+		if err := certWatcher.Start(ctx); err != nil {
+			setupLog.Error(err, "certificate watcher error")
+		}
+	}()
```

Comment:
Similarly, can you move the `certWatcher` creation and start to the `prometheusServer.Start()` method?